clang/lib/CIR/CodeGen/CIRAsm.cpp

```diff
@@ -8,6 +8,20 @@ using namespace cir;
 using namespace clang;
 using namespace mlir::cir;
 
+// The unimplemented features are too local to place them into the
+// UnimplementedFeatureGuarding.h
+struct AsmUnimplemented {
+  static bool Goto() { return false; }
+  static bool LLVMassume() { return false; }
+  static bool unwindClobber() { return false; }
+  static bool memoryEffects() { return false; }
+  static bool vectorType() { return UnimplementedFeature::cirVectorType(); }
+};
```

Comment:
I'd prefer if this could be in `UnimplementedFeatureGuarding.h` since it increases the chance other people could pick up these tasks and continue the work. All these static functions could start with `asm`, example: `asm_goto, asm_llvm_assume, asm_unwind_clobber`, etc.